agent/rpc/middleware/interceptors_test.go

```diff
@@ -266,3 +269,23 @@ func TestRequestRecorder(t *testing.T) {
 		})
 	}
 }
+
+func TestGetNetRPCRateLimitingInterceptor(t *testing.T) {
+	limiter := rate.NewMockRequestLimitsHandler(t)
+
+	logger := hclog.NewNullLogger()
+	var rateLimitInterceptor = GetNetRPCRateLimitingInterceptor(limiter, NewPanicHandler(logger))
+
+	listener, _ := net.Listen("tcp", "127.0.0.1:0")
+
+	t.Run("Allow panics", func(t *testing.T) {
+		limiter.On("Allow", mock.Anything).
+			Panic("uh oh").
+			Once()
+
+		err := rateLimitInterceptor("Status.Leader", listener.Addr())
+
+		require.Error(t, err)
+		require.Equal(t, "rpc: panic serving request", err.Error())
```

Comment:
Doesn't look like we actually use the listener, other than to get an address. Let's hard-code it!

```suggestion
	addr := net.TCPAddrFromAddrPort(netip.MustParseAddrPort("1.2.3.4:5678"))

	t.Run("Allow panics", func(t *testing.T) {
		limiter.On("Allow", mock.Anything).
			Panic("uh oh").
			Once()

		err := rateLimitInterceptor("Status.Leader", addr)

		require.Error(t, err)
		require.Equal(t, "rpc: panic serving request", err.Error())
```

Reply:
Addressed here https://github.com/hashicorp/consul/pull/16022/commits/218ad46ce73cb2b19649c7a264ed66102dba83af